CHANGELOG.md

```diff
@@ -10,6 +10,7 @@
 - Fish is now more forgiving of missing or invalid $TERM values (#3850).
 - The `string` command now supports a `repeat` subcommand with the obvious behavior (#3864).
 - The `string match` command now supports a `--filter` flag to emit the entire string partially matched by a pattern (#3957).
+- The `string replace` command now supports a `--filter` flag to limit output to strings which underwent a replacement (#3348).
```

Comment:
Perhaps `string match --full`, since it prints the full string? Or "--entire-string"?

In retrospect, I wish we had caught this in the initial string discussion, then we could have implemented grep's semantics - full by default, "--only-matching" when necessary.